pkg/diff/diff.go

```diff
@@ -625,6 +627,26 @@ func (sc *Syncer) Solve(ctx context.Context, parallelism int, dry bool, isJSONOu
 		// configuration that is sent to Kong.
 		eventForKong := e
 
+		findLinkedPartials := func(plugin *kong.Plugin) ([]*kong.Partial, error) {
```

Comment:
Could we make this a regular func (and ideally add a UT for it)?

Reply:
I have added a separate function for the same in utils package.
Not sure how to add a UT for this as it requires a kongClient. I would have to yank and put the same logic we use in integration tests for fetching a client. That approach didn't seem well.

Is there any other suggested way? 
I am not a fan for mocking the client. I find that approach not so useful because it's very self-fulfilling. The mock may return the necessary data as we need (as we are the ones adding the mock). So, it can't help in case the actual kong client has started returning different data/no data or stuff like that. So, it doesn't fail!